lib/chef/resource/aws_vpc_peering_connection.rb

```diff
@@ -0,0 +1,73 @@
+require 'chef/provisioning/aws_driver/aws_resource_with_entry'
+
+#
+# An AWS peering connection, specifying which VPC to peer.
+#
+# `name` is not guaranteed unique for an AWS account; therefore, Chef will
+# store the vpc peering connection ID associated with this name in your Chef server in the
+# data bag `data/aws_vpc_peering_connection/<name>`.
+#
+# API documentation for the AWS Ruby SDK for VPC Peering Connections can be found here:
+#
+# - http://docs.aws.amazon.com/sdkforruby/api/Aws/EC2/Types/VpcPeeringConnectionVpcInfo.html
+#
+class Chef::Resource::AwsVpcPeeringConnection < Chef::Provisioning::AWSDriver::AWSResourceWithEntry
+  aws_sdk_type Aws::EC2::VpcPeeringConnection
+  actions :accept, :create, :destroy, :purge, :nothing
+
+  require 'chef/resource/aws_vpc'
+
+  #
+  # The name of this peering connection.
+  #
+  attribute :name, kind_of: String, name_attribute: true
+
+  #
+  # The Local VPC to peer
+  #
+  # May be one of:
+  # - The name of an `aws_vpc` Chef resource.
+  # - An actual `aws_vpc` resource.
+  # - An AWS `VPC` object.
+  #
+  # This is required for new peering connections.
+  #
+  attribute :vpc, kind_of: [ String, AwsVpc, AWS::EC2::VPC ], required: true
```

Comment:
You should remove `required: true` from `vpc` and `vpc_peer` so that someone trying to accept a connection could declare

```ruby
# They could also use the id instead of the resource name
aws_vpc_peering_connection "conn" do
  action: accept
end
```